### PR TITLE
[SYCLomatic][NFC] Add hyperlink for diagnostic message in migration status files

### PR DESCRIPTION
**Pull request by wenhuiNi**
Signed-off-by: Ni, Wenhui <wenhui.ni@intel.com>

### CODE REVIEW

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Comment by zhimingwang36** [comment]
I see the PR is draft,  does it expected to be merged?

**Comment by wenhuiNi** [comment]
The output file still wait doc team to verify whether it working as he expects

File: clang/tools/dpct/generate_api_migration_status.py [hunk 1]
```diff
@@ -5,11 +5,24 @@
...
 keep_only_ask_once = False
 keep_file = True
 
+pattern_re = re.compile("DPCT\d{4}")
+
+def diag_DPCT_link(DPCT_diag:str):
+    return "https://www.intel.com/content/www/us/en/develop/documentation/intel-dpcpp-compatibility-tool-user-guide/top/diagnostics-reference/"+DPCT_diag.lower()+".html"
+
+def format_diagnostic_info_md(DPCT_diag_number:list, is_supported:bool):
+    return " / ".join(("["+diag_number+"]("+diag_DPCT_link(diag_number)+")" for diag_number in DPCT_diag_number if is_supported and diag_number != "DPCT1007"))
```
Comment (zhimingwang36) [inline clang/tools/dpct/generate_api_migration_status.py:20]: DPCT1030 is another unsupport list.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**